awx/main/notifications/grafana_backend.py

```diff
@@ -54,8 +54,8 @@ def __init__(
     ):
         super(GrafanaBackend, self).__init__(fail_silently=fail_silently)
         self.grafana_key = grafana_key
-        self.dashboardId = int(dashboardId) if dashboardId is not None else None
-        self.panelId = int(panelId) if panelId is not None else None
+        self.dashboardId = int(dashboardId) if dashboardId is not None and len(dashboardId) else None
```

Comment:
We would prefer a type check rather than using len() here to ensure the cast to int will be successful.

Reply:
I followed the reproducer in the issue, and it is called out that the value is empty string, `""`. The more standard python way of handling that would be

```diff
diff --git a/awx/main/notifications/grafana_backend.py b/awx/main/notifications/grafana_backend.py
index 0d7dff9e05..7879449526 100644
--- a/awx/main/notifications/grafana_backend.py
+++ b/awx/main/notifications/grafana_backend.py
@@ -54,8 +54,8 @@ class GrafanaBackend(AWXBaseEmailBackend, CustomNotificationBase):
     ):
         super(GrafanaBackend, self).__init__(fail_silently=fail_silently)
         self.grafana_key = grafana_key
-        self.dashboardId = int(dashboardId) if dashboardId is not None else None
-        self.panelId = int(panelId) if panelId is not None else None
+        self.dashboardId = int(dashboardId) if dashboardId else None
+        self.panelId = int(panelId) if panelId else None
         self.annotation_tags = annotation_tags if annotation_tags is not None else []
         self.grafana_no_verify_ssl = grafana_no_verify_ssl
         self.isRegion = isRegion
```

So you use the value as if it were a boolean, it handles false-y values of all kinds, which includes None and empty string.